### PR TITLE
fix: category cooldown commands works in dms like other guild cooldowns

### PR DESCRIPTION
**Pull request by Revnoplex**
## Summary
While testing certain cooldown types for my cooldown handler. I stumbled upon an error when using a command that had a category cooldown in DMs. Note that the other guild-like cooldown types such as `BucketType.guild`, `BucketType.member` and `BucketType.role` work fine in DMs. So it is reasonable to consider this an unintended error rather than intentional and should be fixed. The error was caused by returning the bucket id using `(msg.channel.category or msg.channel).id` which can result in an error when it tries to access the `category` attribute when the channel is not a `GuildChannel`  such as a `PrivateChannel` which doesn't have a `category` attribute so it results in a `AttributeError`. By using `(msg.channel.category if isinstance(msg.channel, discord.abc.GuildChannel) else msg.channel).id` (a `GuildChannel` always has a category attribute), it won't try to access the attribute when it can't.

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

File: discord/ext/commands/cooldowns.py [hunk 1]
```diff
@@ -69,7 +70,11 @@ def get_key(self, msg: Message) -> Any:
         elif self is BucketType.member:
             return (msg.guild and msg.guild.id), msg.author.id
         elif self is BucketType.category:
-            return (msg.channel.category or msg.channel).id  # type: ignore
+            return (
+                msg.channel.category
+                if isinstance(msg.channel, discord.abc.GuildChannel)
+                else msg.channel
+            ).id
```
Comment (Dorukyum) [inline discord/ext/commands/cooldowns.py:77]:
```suggestion
            return (
                msg.channel.category.id
                if isinstance(msg.channel, discord.abc.GuildChannel)
                else msg.channel.id
            )
```

File: discord/ext/commands/cooldowns.py [hunk 2]
```diff
@@ -69,7 +70,11 @@ def get_key(self, msg: Message) -> Any:
         elif self is BucketType.member:
             return (msg.guild and msg.guild.id), msg.author.id
         elif self is BucketType.category:
-            return (msg.channel.category or msg.channel).id  # type: ignore
+            return (
+                msg.channel.category.id
+                if isinstance(msg.channel, discord.abc.GuildChannel)
```
Comment (Dorukyum) [inline discord/ext/commands/cooldowns.py:75]: Hold on, `category` can still be `None`.

```suggestion
                if isinstance(msg.channel, discord.abc.GuildChannel)
                and channel.category
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

File: discord/ext/commands/cooldowns.py (hunk 2 above)
Comment (Revnoplex) [reply discord/ext/commands/cooldowns.py:75]: oh it should of been **msg**.channel.category

**Comment by Revnoplex** [comment]
no, wait it undid it

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**